src/actions/deviceActions.js

```diff
@@ -470,10 +470,15 @@ export function setDeviceRunning(isRunning) {
 
 export function setPowerMode(isSmuMode) {
     return async dispatch => {
-        await device.ppkSetPowerMode(isSmuMode ? 2 : 1);
+        await device.ppkSetPowerMode(isSmuMode);
```

Comment:
I thought sending the command to the device must change order with the running state.
```js
  if (isSmuMode) {
    dispatch(setDeviceRunning(false));
    await device.ppkSetPowerMode(true);
    dispatch(setPowerModeAction(true)); // set to source mode
  } else {
    await device.ppkSetPowerMode(false);
    dispatch(setPowerModeAction(false)); // set to ampere mode
    dispatch(setDeviceRunning(true));
  }
```

Reply:
Oh right, oops